src/foam/u2/crunch/CrunchController.js

```diff
@@ -66,183 +63,19 @@ foam.CLASS({
-                    this.EQ(this.AgentCapabilityJunction.EFFECTIVE_USER, this.subject.user.id)
-                  )
-                ),
-                this.EQ(this.UserCapabilityJunction.TARGET_ID, cap.id))
-            ).then((ucj) => {
-              // TODO: should be calling save
-              if ( ucj == null ) {
-                ucj = this.UserCapabilityJunction.create({
-                  sourceId: associatedEntity.id,
-                  targetId: cap.id
-                });
-              }
-              this.userCapabilityJunctionDAO.put(ucj).then(() => console.log('SAVED (no-data cap)', cap.id));
-            });
-          }
-        )).then(() => {
-          wizardResolve();
-        });
-      });
-    },
-
-    async function launchWizard(capability) {
-      if ( typeof capability == 'string' ) capability = await this.capabilityDAO.find(capability);
-      var associatedEntity = capability.associatedEntity === foam.nanos.crunch.AssociatedEntity.USER ? this.subject.user : this.subject.realUser;
-      var ucj = await this.userCapabilityJunctionDAO.find(
-        this.AND(
-          this.OR(
-            this.AND(
-              this.NOT(this.INSTANCE_OF(this.AgentCapabilityJunction)),
-              this.EQ(this.UserCapabilityJunction.SOURCE_ID, associatedEntity.id)
-            ),
-            this.AND(
-              this.INSTANCE_OF(this.AgentCapabilityJunction),
-              this.EQ(this.UserCapabilityJunction.SOURCE_ID, associatedEntity.id),
-              this.EQ(this.AgentCapabilityJunction.EFFECTIVE_USER, this.subject.user.id)
-            )
-          ),
-          this.EQ(this.UserCapabilityJunction.TARGET_ID, capability.id)
-        )
-      );
-      if ( ucj ) {
-        var statusGranted = ucj.status === this.CapabilityJunctionStatus.GRANTED;
-        var statusPending = ucj.status === this.CapabilityJunctionStatus.PENDING 
-          || ucj.status === this.CapabilityJunctionStatus.APPROVED;
-        if ( statusGranted || statusPending ) {
-          var message = statusGranted ? this.CANNOT_OPEN_GRANTED : this.CANNOT_OPEN_PENDING;
-          this.ctrl.notify(message, '', this.LogLevel.INFO, true);
-          return;
-        }
-        var nothingTodo = ucj.status === this.CapabilityJunctionStatus.ACTION_REQUIRED;
-        return this.startWizardFlow(capability.id, nothingTodo);
-      }
-      return this.startWizardFlow(capability.id, false);
+    function createWizardSequence(capabilityOrId) {
+      return this.Sequence.create(null, this.__subContext__.createSubContext({
+        rootCapability: capabilityOrId
+      }))
+        .add(this.CapabilityAdaptAgent)
+        .add(this.LoadCapabilitiesAgent)
+        .add(this.CheckPendingAgent)
+        .add(this.CreateWizardletsAgent)
+        .add(this.RequirementsPreviewAgent)
+        .add(this.StepWizardAgent)
+        .add(this.PutFinalJunctionsAgent)
+        .add(this.TestAgent)
+        ;
```

Comment:
weird semi colon

Reply:
It prevents merge conflicts if two people edit this list at the same time